core/src/Kokkos_View.hpp

```diff
@@ -1536,7 +1536,11 @@ class View : public ViewTraits<DataType, Properties...> {
       : View(arg_prop,
              typename traits::array_layout(arg_N0, arg_N1, arg_N2, arg_N3,
                                            arg_N4, arg_N5, arg_N6, arg_N7),
-             check_input_args::yes) {}
+             check_input_args::yes) {
+    static_assert(traits::array_layout::is_extent_constructible,
+                  "Layout is not extent constructible. Use overload taking a "
```

Comment:
Please suggest a message that will be easier to understand for users

Reply:
I don't know, seems pretty obvious to me?

Reply:
"is not constructible from extent arguments.  Use ..."

Reply:
I would just use the second sentence "Use the overload taking a layout object instead!".